### PR TITLE
BAH-913 | Sowmika | Reset appointment status

### PR DESCRIPTION
**Pull request by sowmika148**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java [hunk 1]
```diff
@@ -172,26 +172,37 @@ public Appointment getAppointmentByUuid(String uuid) {
...
             createEventInAppointmentAudit(appointment, notes);
-        }
-        else {
+        } else {
             String message = StringUtils.join(errors, "\n");
             throw new APIException(message);
         }
     }
 
+    private void throwExceptionIfNoProperPrivileges(Appointment appointment, AppointmentStatus appointmentStatus) {
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java]: Can we have a better name? The function should not be named "throw*". Should be named for intent - e.g. What are you validating - validateUserPrivilege() .. or something like that.

File: api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImplTest.java [hunk 2]
```diff
@@ -426,7 +427,7 @@ public void shouldThrowExceptionOnAppointmentSaveIfUserHasOnlyOwnPrivilegeAndPro
     public void shouldThrowExceptionOnAppointmentStatusChangeIfUserHasOnlyOwnPrivilegeAndProviderAndUserIsNotTheSamePerson() {
         setupForOwnPrivilegeAccess(exceptionCode);
         expectedException.expect(APIAuthenticationException.class);
-        appointmentsService.changeStatus(appointment, null, null);
+        appointmentsService.changeStatus(appointment, "Scheduled", null);
```
[the remark this answers is not in the document]
Comment (sowmika148) [reply api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImplTest.java]: Changed

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java [hunk 3]
```diff
@@ -38,8 +38,9 @@
 @Transactional
 public class AppointmentsServiceImpl implements AppointmentsService {
 
-    private static final int EMPTY_SET_SIZE = 0;
     private Log log = LogFactory.getLog(this.getClass());
+    private static final String RESET_APPOINTMENT_STATUS_PRIVILEGE = "Reset Appointment Status";
```
[the remark this answers is not in the document]
Comment (sowmika148) [reply api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java]: Moved

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java [hunk 4]
```diff
@@ -87,30 +88,29 @@ public Appointment validateAndSave(Appointment appointment) throws APIException
             throw new APIAuthenticationException(Context.getMessageSourceService().getMessage("error.privilegesRequired",
                     new Object[] { MANAGE_APPOINTMENTS }, null));
         }
-        if(!CollectionUtils.isEmpty(appointmentValidators)){
+        if (!CollectionUtils.isEmpty(appointmentValidators)) {
```
[the remark this answers is not in the document]
Comment (sowmika148) [reply api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java]: This validation is at service layer - api

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java (hunk 1 above)
Comment (sowmika148) [reply api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java]: Done

**Pull request merged**